src/Services/Azure/Authentication/CustomChainedCredential.cs

```diff
@@ -119,4 +135,76 @@ private static DefaultAzureCredential CreateDefaultCredential(string? tenantId)
 
         return new DefaultAzureCredential(defaultCredentialOptions);
     }
+
+    private static TokenCredential? CreateVsCodeBrokerCredential(string? tenantId)
+    {
+        const string vsCodeClientId = "aebc6443-996d-45c2-90f0-388ff96faa56";
+        string userProfile = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+        string authRecordPath = Path.Combine(userProfile, ".azure", "ms-azuretools.vscode-azureresourcegroups", "authRecord.json");
+        if (!File.Exists(authRecordPath))
+        {
+            // Try .Azure if .azure is not present
+            authRecordPath = Path.Combine(userProfile, ".Azure", "ms-azuretools.vscode-azureresourcegroups", "authRecord.json");
+            if (!File.Exists(authRecordPath))
+            {
+                return null;
+            }
+        }
+
+        AuthenticationRecord? authRecord;
+        try
+        {
+            using var stream = File.OpenRead(authRecordPath);
+            authRecord = AuthenticationRecord.Deserialize(stream);
+        }
+        catch
+        {
+            // Deserialization failed
+            return null;
+        }
+
+        if (authRecord is null)
+            return null;
+
+        // Validate client ID
+        if (!string.Equals(authRecord.ClientId, vsCodeClientId, StringComparison.OrdinalIgnoreCase))
+            return null;
+
+        // Validate tenant ID if present
+        if (!string.IsNullOrEmpty(authRecord.TenantId) && !IsValidTenantId(authRecord.TenantId))
+            return null;
+
+        // Prefer explicit tenantId, else use from auth record
+        string? effectiveTenantId = !string.IsNullOrEmpty(tenantId)
+            ? tenantId
+            : authRecord.TenantId;
+
+        var options = new InteractiveBrowserCredentialBrokerOptions(0)
+        {
+            ClientId = vsCodeClientId,
+            TenantId = effectiveTenantId,
+            AuthenticationRecord = authRecord
+        };
+
+        return new InteractiveBrowserCredential(options);
+    }
+
+    private static bool IsValidTenantId(string tenantId)
+    {
+        foreach (char c in tenantId)
+        {
+            if (!IsValidTenantCharacter(c))
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    // Helper to validate tenant ID: only hex digits and dashes allowed
+    private static bool IsValidTenantCharacter(char c)
+    {
+        return (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || (c >= '0' && c <= '9') || (c == '.') || (c == '-');
```

Comment:
>c <= 'z' [](http://example.com/codeflow?start=28&length=8)

Only A through F are valid for hex. Why do we check more than that?

Reply:
the description comment is incorrect with only hex, removing it.